src/ucp/wireup/wireup.c

```diff
@@ -1343,25 +1343,165 @@ static void ucp_wireup_discard_uct_eps(ucp_ep_h ep, uct_ep_h *uct_eps,
+{
+    ucp_ep_config_key_t key;
+    unsigned addr_indices[UCP_MAX_LANES];
+
+    ucp_ep_config_key_reset(&key);
+
+    /* Perform selection in order to initialize configuration key */
+    if (ucp_wireup_select_lanes(ep, ep_init_flags, *tl_bitmap, remote_address,
+                                addr_indices, &key, 1) != UCS_OK) {
+        return 0;
+    }
+
+    return ucp_wireup_can_reconfigure_internal(ep, &key, remote_address,
+                                               addr_indices);
+}
+
+static ucp_lane_index_t
+ucp_wireup_find_non_reused_lane(ucp_ep_h ep, const ucp_ep_config_key_t *key)
+{
+    if (ucp_ep_has_cm_lane(ep)) {
+        return key->cm_lane;
+    }
+
+    /* Just use first lane, as only non-reused lanes are allowed at the
+     * moment. */
+    ucs_assert(key->num_lanes > 0);
+    return 0;
+}
+
+static ucs_status_t ucp_wireup_replace_wireup_msg_lane(ucp_ep_h ep,
+                                                       ucp_ep_config_key_t *key,
+                                                       uct_ep_h *new_uct_eps)
+{
+    ucp_lane_index_t old_lane, new_lane;
+    ucp_wireup_ep_t *old_ep, *new_ep;
+    uct_ep_h uct_ep;
+    ucp_rsc_index_t aux_rsc_index;
+    int is_p2p;
+    ucs_status_t status;
+
+    /* Get old wireup lane */
+    old_lane = ucp_wireup_get_msg_lane(ep, UCP_WIREUP_MSG_REQUEST);
+    old_ep   = ucp_wireup_ep(ucp_ep_get_lane(ep, old_lane));
+    ucs_assert(old_ep != NULL);
+
+    /* Set wireup EP for new configuration's wireup lane */
+    if (ucp_ep_has_cm_lane(ep)) {
+        /* Use existing EP from CM lane */
+        new_ep = ucp_ep_get_cm_wireup_ep(ep);
+        ucs_assert(new_ep != NULL);
+        aux_rsc_index = ucp_ep_get_rsc_index(ep, old_lane);
+        is_p2p        = ucp_ep_is_lane_p2p(ep, old_lane);
+    } else {
+        /* Create new EP for non-CM flow */
+        status = ucp_wireup_ep_create(ep, &uct_ep);
+        if (status != UCS_OK) {
+            return status;
+        }
+        new_ep        = ucp_wireup_ep(uct_ep);
+        aux_rsc_index = ucp_wireup_ep_get_aux_rsc_index(&old_ep->super.super);
+        is_p2p        = old_ep->flags & UCP_WIREUP_EP_FLAG_AUX_P2P;
+    }
+
+    /* Move aux EP to new wireup lane */
+    ucs_assert(aux_rsc_index != UCP_NULL_RESOURCE);
+    ucp_wireup_ep_set_aux(new_ep, ucp_wireup_ep_extract_msg_ep(old_ep),
+                          aux_rsc_index, is_p2p);
+
+    /* Remove old wireup_ep as it's not needed anymore.
+     * NOTICE: Next two lines are intentionally not merged with the lane
+     * removal loop in ucp_wireup_check_config_intersect, because of future
+     * support for non-wireup EPs reconfiguration (which will modify this
+     * code). */
+    uct_ep_destroy(&old_ep->super.super);
+    ucp_ep_set_lane(ep, old_lane, NULL);
+
+    /* Select CM/non-reused lane as new wireup lane */
+    new_lane              = ucp_wireup_find_non_reused_lane(ep, key);
+    new_uct_eps[new_lane] = &new_ep->super.super;
```

Comment:
I think it may complicate the flow, because you need to return [new_lane,new_ep] from this function and then add extra "if" block in `ucp_wireup_check_config_intersect` in order to properly initialize `new_uct_eps`.

And anyway the value of `new_uct_eps` will not be overwritten, because in `ucp_wireup_check_config_intersect` we only handle "reused" lanes.